src/main/java/org/jabref/logic/formatter/bibtexfields/CleanupURLFormatter.java

```diff
@@ -0,0 +1,67 @@
+package org.jabref.logic.formatter.bibtexfields;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.cleanup.Formatter;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Cleanup URL link
+ */
+
+public class CleanupURLFormatter implements Formatter {
+
+    private static final Log LOGGER = LogFactory.getLog(CleanupURLFormatter.class);
+
+    @Override
+    public String getName() {
+        return Localization.lang("Cleanup URL Link");
+    }
+
+    @Override
+    public String getKey() {
+        return "cleanup_url";
+    }
+
+    @Override
+    public String format(String value) {
+        URLDecoder urlDecoder = new URLDecoder();
+
+        String decodedLink = value;
+        String toDecode = value;
+
+        // This regexp find "url=" or "to=" parameter in full link and get text after them
+        String urlRegexp = "(?:url|to)=([^&]*)";
```

Comment:
Extract the pattern to a static Pattern and use Pattern.compile
This is more efficient, I think examples are in other integrity checker classes